keanu-project/src/main/java/io/improbable/keanu/util/status/RemainingTimeComponent.java

```diff
@@ -0,0 +1,35 @@
+package io.improbable.keanu.util.status;
+
+import java.time.Duration;
+
+/**
+ * {@link StatusBarComponent} that shows the estimated remaining time for algorithm completion. Also shows average step
+ * time and elapsed time.
+ */
+public class RemainingTimeComponent extends TimeComponent {
+
+    private final AverageTimeComponent averageTime = new AverageTimeComponent();
+    private final long totalSteps;
+
+    public RemainingTimeComponent(long totalSteps) {
+        this.totalSteps = totalSteps;
+    }
+
+    @Override
+    public String render() {
```

Comment:
It looks like this sometimes leaves a non-zero amount of time remaining left even after it's finished. Can we either not print the time remaining when it's done or zero it out?